Source/RapyutaSimulationPlugins/Private/Core/RRBaseActor.cpp

```diff
@@ -43,19 +42,26 @@ void ARRBaseActor::PreInitializeComponents()
 {
     Super::PreInitializeComponents();
 
-    // SETUP + CONFIGURE ESSENTIAL GAME & SIM COMMON OBJECTS
+    // SETUP + CONFIGURE GAME & SIM COMMON OBJECTS
     // NOTE: These are used ONLY for dynamic runtime actors.
     // In child BP actors, except for GameSingleton, other Game framework entities maybe not available yet
-    GameMode = URRCoreUtils::GetGameMode<ARRGameMode>(this);
-    GameState = URRCoreUtils::GetGameState<ARRGameState>(this);
+#if RAPYUTA_USE_SCENE_DIRECTOR
+    RRGameState = URRCoreUtils::GetGameState<ARRGameState>(this);
+    check(RRGameState)
+#endif
+        RRROS2GameMode = URRCoreUtils::GetGameMode<ARRROS2GameMode>(this);
```

Comment:
Since ARRGameMode is still used for `IsSimType()` or `IsDataSynthSimType()` in DataSynthApp, please keep
`RRGameMode = URRCoreUtils::GetGameMode<ARRGameMode>(this);` without check or verify

Reply:
Where I can find?

Reply:
I mentioned it [here](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/pull/154#discussion_r1181502911)
Also a quick Ctrl+Shift+F will show it like [here](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Source/RapyutaSimulationPlugins/Private/Core/RRProceduralMeshComponent.cpp#L122)